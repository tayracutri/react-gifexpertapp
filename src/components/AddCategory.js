import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState('')
    const handleInput = (e) => {
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {

            setCategories(cat => [inputValue, ...cat]);
            setinputValue('');
        }
    }

    return (

        <form onSubmit={handleSubmit}>

            <input type='text'
                value={inputValue}
                onChange={handleInput}
            />

        </form>

    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
