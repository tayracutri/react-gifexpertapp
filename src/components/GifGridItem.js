import React from 'react';



export const GifGridItem = ({ title, url }) => {


    return (
        <div className='card animate__animated fadeIn'>

            <img src={url} alt={title} />
            <p> {title}</p>


        </div >)

}
