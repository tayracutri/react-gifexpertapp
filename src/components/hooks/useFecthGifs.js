import { useState, useEffect } from 'react'
import { getGifs } from '../helpers/GetGifs';




export const useFecthGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs(category)
            .then(imgs => {
                setTimeout(() => {
                    setState({
                        data: imgs,
                        loading: false
                    })
                }, 100);
            })
    }, [category])




    return state;
};