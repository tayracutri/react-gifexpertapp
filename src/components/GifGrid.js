import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types'
import { useFecthGifs } from './hooks/useFecthGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data, loading } = useFecthGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeInDown"> {category} </h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="grid-card">

                {data.map(img => (
                    <GifGridItem key={img.id} {...img} />

                ))
                }
            </div>

        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}