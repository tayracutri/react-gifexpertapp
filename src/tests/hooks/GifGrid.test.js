import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFecthGifs } from '../../components/hooks/useFecthGifs';
import '@testing-library/jest-dom';
jest.mock('../../components/hooks/useFecthGifs')





describe('test on GifGrid', () => {
    const category = 'Dani';


    test('should show istelf correctly', () => {

        useFecthGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />)

        expect(wrapper).toMatchSnapshot();
    });


    test('should show itmes after img being uploaded', () => {


        const data = [{
            id: 1,
            url: 'htpps://google.com',
            title: 'Dani'
        },
        {
            id: 2,
            url: 'htpps://facebook.com',
            title: 'any title'
        },
        ]

        useFecthGifs.mockReturnValue({
            data,
            loading: false
        });


        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(data.length)


    });

});
