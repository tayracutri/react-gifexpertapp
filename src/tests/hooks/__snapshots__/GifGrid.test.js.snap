// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on GifGrid should show istelf correctly 1`] = `ShallowWrapper {}`;

exports[`test on GifGrid should show itmes after img being uploaded 1`] = `ShallowWrapper {}`;
